### PR TITLE
Some files for debugging via VSCode

### PR DESCRIPTION
**Pull request by grbd**
I've added some files for debugging under Visual Studio Code
This should allow for stepping through the code / breakpoints etc

I'd recommend the following to use it

  * LLVM (For the LLDB Debugger) installed and on the path
  * VSCode Extension - CodeLLDB
  * VSCode Extension - "rust-analyser" (not the "rust" one)
  * VSCode Extension - "Task Explorer"
  * VSCode Extension - "crates"

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ dump.rdb
 /rs*.sh
 /docs/_site
 /core
+/.vscode
```
Comment (sagiegurari) [inline .gitignore:12]: i think we can remove this entry.

File: README.md [hunk 2]
```diff
@@ -113,6 +113,7 @@
 * [Roadmap](#roadmap)
 * [Editor Support](#editor-support)
 * [Contributing](.github/CONTRIBUTING.md)
+* [Development](#development)
```
Comment (sagiegurari) [inline README.md:116]: this is a generated file. you can look at the doc folder, and you will see a sub folder with templates for the nav and content md files.
so this goes to the nav.
also, we should not add a new section, instead lets use the Editor Support section that already exists and talks about vim.

File: README.md [hunk 3]
```diff
@@ -3477,6 +3478,32 @@ You can view the future development items list in the [github project issues](ht
 ## Contributing
 See [contributing guide](.github/CONTRIBUTING.md)
 
+<a name="development"></a>
+## Development
+
+### VSCode
+
+For debugging purposes there are some example .vscode files located within the [docs/vscode-example](./docs/vscode-example) directory
```
Comment (sagiegurari) [inline README.md:3486]: awesome!

File: README.md [hunk 4]
```diff
@@ -3477,6 +3478,32 @@ You can view the future development items list in the [github project issues](ht
 ## Contributing
 See [contributing guide](.github/CONTRIBUTING.md)
 
+<a name="development"></a>
+## Development
```
Comment (sagiegurari) [inline README.md:3482]: lets move this whole thing under 'editor support' and maybe create sub section for vim and vscode (vim already exists).

**Review by sagiegurari — CHANGES_REQUESTED** [review]
you put the vscode files in the right place.
the docs, you need to modify the md files (content and nav md files) which are used
1. in the github.io site as is
2. used to generate the full readme md

**Comment by sagiegurari** [comment]
@grbd just want to make sure you saw the comments in this PR

**Comment by sagiegurari** [comment]
@grbd friendly ping :) 
you don't really need jekyl for this PR. just look at the generated md files.

File: .gitignore (hunk 1 above)
Comment (grbd) [reply .gitignore]: This should now be done

**Comment by grbd** [comment]
Ok I think I've resolved all the above now

File: docs/vscode-example/tasks.json [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            "detail": "Clean the target build directory",
+            "type": "process",
+            "group": "build",
+            "command": "cargo",
+            "args": [
+                "make",
+                "clean"
+            ]
+        },
+        {
```
Comment (sagiegurari) [inline docs/vscode-example/tasks.json:37]: i'll change this one to use the cargo-make formatter task.
it comes with built in support to auto install rustfmt

File: docs/_includes/content.md [hunk 6]
```diff
@@ -3355,8 +3355,33 @@ You can view the future development items list in the [github project issues](ht
...
+### Rust Format
+
+After making changes to the code, rustfmt can be used to automatically reformat the code within the repository
+```
+# To install
+rustup component add rustfmt
+# To run on the cargo project
+cargo fmt
+```
+
```
Comment (sagiegurari) [inline docs/_includes/content.md:3384]: i'll remove this one as cargo make comes with built in formatting task that auto installs it for you

File: docs/vscode-example/tasks.json [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            "label": "run the tests",
+            "detail": "Run the tests",
+            "type": "process",
+            "group": "build",
+            "command": "cargo",
+            "args": [
+                "test"
+            ]
+        },
```
Comment (sagiegurari) [inline docs/vscode-example/tasks.json:56]: i'll also change this one to use the cargo make test which knows to run multi phase tests (tests for single thread + multi thread + custom tasks for tests)

**Review by sagiegurari — APPROVED** [review]

**Comment by sagiegurari** [comment]
thanks @grbd 
i'll be doing changes after merge to use the cargo-make tasks and not cargo commands as they do a lot more (including auto install rustfmt, run multi phase tests and so on...)

**Comment by grbd** [comment]
Ok, the bit I added about using cargo fmt
was just a reminder to use cargo fmt after changing any code to check for any formatting issues before submitting changes to the cargo-make repo.

**Comment by sagiegurari** [comment]
@grbd thanks for the PR. i'm merging it and i'll make few changes as i commented on the files themselves.

**Pull request merged**